fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/MergePercentileToArray.java

```diff
@@ -76,19 +75,19 @@ public Rule build() {
     // Merge percentile into percentile_array according to funcMap
     private List<AggregateFunction> getPercentileArrays(Map<DistinctAndExpr, List<AggregateFunction>> funcMap) {
         List<AggregateFunction> newPercentileArrays = Lists.newArrayList();
+
         for (Map.Entry<DistinctAndExpr, List<AggregateFunction>> entry : funcMap.entrySet()) {
-            List<Literal> literals = new ArrayList<>();
+            List<Expression> literals = new ArrayList<>();
             for (AggregateFunction aggFunc : entry.getValue()) {
-                List<Expression> literal = aggFunc.child(1).collectToList(expr -> expr instanceof Literal);
-                literals.add((Literal) literal.get(0));
+                literals.add(aggFunc.child(1));
             }
-            ArrayLiteral arrayLiteral = new ArrayLiteral(literals);
```

Comment:
if all expression are literal, generate a array literal directly for better performance